ai-service/src/ai_service/handlers/ingest.py

```
@@ -0,0 +1,71 @@
+        for file_path in code_files:
+            try:
+                with open(file_path, encoding="utf-8") as f:
+                    code = f.read().strip()
+                    if not code:
+                        logger.warning(f"Skipping empty file: {file_path}")
+                        continue
+                    code_snippets.append(code)
+                    embeddings.append(embedder.embed_text(code))
+            except FileNotFoundError:
+                err = errors.FileReadError.file_not_found(file_path)
+                logger.error(err)
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)